common/protob/messages-cardano.proto

```diff
@@ -210,6 +215,9 @@ message CardanoSignTxInit {
     optional bytes script_data_hash = 16;
     required uint32 collateral_inputs_count = 17;
     required uint32 required_signers_count = 18;
+    optional bool has_collateral_return = 19 [default=false];
```

Comment:
The `has_collateral_return`, `reference_inputs_count`, `CardanoTxOutput.format`, `CardanoTxOutput.inline_datum_size` and ``CardanoTxOutput.reference_script_size` fields should ideally be `required`.

But we're worried whether Connect will be released in time with the FW because of the new major Connect version and migration to the trezor-suite repo. So we rather made the fields optional with default values because otherwise the new FW would not work with old Connect since it would be expecting these fields.

What do you think about this please?

Reply:
Adding all new fields as optionals with defaults is correct IMHO, unless the field is actually required for some part of the logic. If the fields are functionally optional (say for older transaction type), AND if there is a default value that makes sense (such as `has_collateral_return = false` by default, similarly `*_count` added in the past can be implicitly 0), it's good that they are optional and older clients not using these features cleanly continue working.

That said, `total_collateral` is optional but has no default value. Would it make sense to set 0? Can a valid transaction be produced with 0 collateral?

Reply:
or is `total_collateral` intended as optional?

Reply:
`total_collateral` is intended as optional so that it is left out of the transaction completely if it's not set i.e. 0 is probably not a valid default. I would expect e.g. ordinary transactions with `total_collateral` set to 0 to be invalid but I haven't verified this.